src/test/scala/scorex/core/network/NetworkControllerSpec.scala

```diff
@@ -306,6 +308,109 @@ class NetworkControllerSpec extends NetworkTests {
     system.terminate()
   }
 
+  it should "skip connection attempt if the only connection in PeerManager is marked as unconfirmed" in {
```

Comment:
I think we need to add a tricky case test for `getPeerAddress` function we started to use.
Which for the description is:
"Returns local address of peer for local connections and WAN address of peer for external connections. When local address is not known, try to ask it at the UPnP gateway"

So let's emulate local and external connections.

Reply:
Added test to emulate the local address peer case, the other tests were already passing through the external connection branch condition